---
title: "Data Analysis"
output: html_document
date: "2024-04-08"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Differential Phasing between Circadian Clocks in the Brain and Peripheral Organs in Humans
## Jacob J. Hughey and Atul J. Butte 2016
### Background 
The circadian clock, an approximate 24-hour molecular oscillator found in all species to date, regulates physiological rhythms such as sleep/wake, hunger, metabolism, and DNA repair. 

Each organism possesses multiple circadian clocks, with a "master" clock in the SCN receiving light cues from the retina. Peripheral clocks in organs respond to both SCN signals and environmental cues like feeding schedules.

Alignment of these clocks is crucial for proper circadian function. Environmental factors can disrupt this alignment, leading to negative health consequences. A disrupted "clock" has been associated with different health conditions such as cancer, cognitive disorders (e.g., Alzheimer's disease), and mood disorders (e.g., major depressive disorder). While core clock genes are conserved in mammals, the dynamics of clock interactions may vary.

*The goal of this study is to compare (circadian) gene expression data across organs in humans and other mammals to shed light on circadian clock phases and their variations.*

### Data Set Collection
The authors collected 15 datasets and 13 studies from human, mice, rats, and other diurnal/nocturnal mammals. 
They made sure to only include control subjects from human studies (i.e., no data from individuals with major depressive disorder). 
Authors mentioned a supplemental data set that describes all data sets, but this could not be found. The file that contains the data also does not have the supplemental information...

### Data Analysis 
Circadian rhythm data is unique in that it is reported in time from when the lights are on (zeitgeiber time zero, ZT0) or when the sun rises (circadian time 0, CT0). Time collected from the data were converted accordingly. 

For each data set, authors fit a periodic smoothing ... 


#### Figure Analysis: Figure 1
For Hughey & Butte's **Figure 1**, they look at circadian gene expression of three "clock" genes in human blood and brain. For context, the temporal oscillation of many genes are controlled by the circadian feedback loop of transcriptional activators (e.g., CLOCK and BMAL1) and repressors (e.g., PER, CRY, and REV-ERB). Without these genes, the internal clock cannot run properly. As such, the authors investigated ARNTL (also known as BMAL1), NR1D1 (also known as REVERB) and PER3. 

Each point is a sample and the shape indicates the dataset. In each plot, the blak lie shows the periodic smoothing spline corresponding to the mean expression over time. Expression values for each gene in each dataset were scaled to have mean of zero and an SD of 1. 

```{r, Fig 1}
# Load in libraries
library(tidyverse)
library(skimr)
library(dplyr)

# In order to download biobase if R is v4.0 or later ->
# install.packages("BiocManager")
# BiocManager::install("Biobase")
library(Biobase)

# Get circadian genes from github 
# This file contains species name, entrez code, and symbol
f_ClockGenes <- "https://raw.githubusercontent.com/kierstenbell/data-analysis-rep/main/clock_genes.csv"
ClockGenes <- read_csv(f_ClockGenes, col_names = TRUE)
skim(ClockGenes)
# Change entrez entry from numeric to character
ClockGenes <- ClockGenes %>%
  mutate(entrez = as.character(entrez))

# Get study meta data
f_MetaData <- "https://raw.githubusercontent.com/kierstenbell/data-analysis-rep/main/study_metadata.csv"
MetaData <- read_csv(f_MetaData, col_names = TRUE)
head(MetaData)
# rownames(StudyMetaData) <- StudyMetaData[,'study'] # why are we adding another column with idential information to study


# Get human metadata from my github
# Study = data from Gene Expression Omniobus (GEO) study number (ex: GSE45642)
# Title = subject ID with more information
# Sample = GEO accession
# Subject = subject ID
f <- "https://raw.githubusercontent.com/kierstenbell/data-analysis-rep/main/sample_metadata_human.csv"
d_HumanMetaData <- read_csv(f, col_names = TRUE)

# Use skim from {skimr} to get idea of data and see how many/where missing n 
skim(d_HumanMetaData) 
head(d_HumanMetaData)

# Need biobase to read this document
# data_human.rds is 227 MB.  Need to figure out how to connect to github
d_human <- readRDS("/Users/Kiersten/Documents/Academic Life/Classes/2024_01 Spring /Repos/Hughey & Butte/data_human.rds")
head(d_human)
# d_human emat is 150 MB... to upload to github you will have to use command line?
d_human_emat <- readRDS("/Users/Kiersten/Documents/Academic Life/Classes/2024_01 Spring /Repos/Hughey & Butte/data_human_emat.rds")

# Authors created a function called loadSpeciesData 
loadSpeciesData = function(species, studyMetadata) {
	species$sampleMetadata = read.csv(species$sampleMetadataFileName, stringsAsFactors=FALSE)
	rownames(species$sampleMetadata) = species$sampleMetadata[,'sample']
	if (species$denovo) {
		studyMetadataNow = studyMetadata[studyMetadata[,'study'] %in% unique(species$sampleMetadata[,'study']),]
		species$esetList = getStudyDataList(species$parentFolderPath, studyMetadataNow)
		saveRDS(species$esetList, file=paste0(species$metaAnalysisName, '.rds'))
	} else {
		species$esetList = readRDS(paste0(species$metaAnalysisName, '.rds'))}
	return(species)}

human = list(metaAnalysisName = 'd_human',
				 sampleMetadataFileName = 'https://raw.githubusercontent.com/kierstenbell/data-analysis-rep/main/sample_metadata_human.csv',
				 parentFolderPath = '.',
				 denovo = FALSE)
```

```{r}
human = loadSpeciesData(human, MetaData)


```
